apps/meteor/app/api/server/v1/users.ts

```diff
@@ -393,10 +393,16 @@ API.v1.addRoute(
 		async get() {
 			const { fields } = await this.parseJsonQuery();
 
-			const user = await getFullUserDataByIdOrUsername(this.userId, {
-				filterId: (this.queryParams as any).userId,
-				filterUsername: (this.queryParams as any).username,
-			});
+			const searchTerms: [string, 'id' | 'username' | 'importId'] | false =
+				('userId' in this.queryParams && !!this.queryParams.userId && [this.queryParams.userId, 'id']) ||
+				('username' in this.queryParams && !!this.queryParams.username && [this.queryParams.username, 'username']) ||
+				('importId' in this.queryParams && !!this.queryParams.importId && [this.queryParams.importId, 'importId']);
```

Comment:
Would it be the same to use ajv like this https://stackoverflow.com/a/48778838 ?

We validate they're not empty when present, and on here, you can just
`searcterms = this.queryparams.userid || username || importid`

Reply:
In this case other than the value I also need to know which attribute the value came from, so that wouldn't work. Ajv is already validating the param structure.